tests/integration/security/rbac/util/rbac_util.go

```diff
@@ -56,40 +62,21 @@ func (tc TestCase) Check() error {
 			err = resp.CheckOK()
 		}
 		if err != nil {
-			return fmt.Errorf("%s to %s:%s%s using %s: expected allow but was denied: %v",
-				req.From.Config().Service,
-				req.Options.Target.Config().Service,
-				req.Options.PortName,
-				req.Options.Path,
-				req.Options.Scheme,
-				err)
+			return getError(req, "allow with code 200", fmt.Sprintf("error: %v", err))
 		}
 	} else {
-		if err != nil && !strings.Contains(err.Error(), "EOF") {
```

Comment:
@nmittler 
This is the part for fixing a potential issue. This part we expect to see the TCP request gets denied. It should error either `err` is nil or `err` doesn't include "EOF"